Src/Model/ApiModel.php

```diff
@@ -0,0 +1,55 @@
+<?php
+namespace Aot\Model;
+
+class ApiModel
+{
+    private $format = [];
+
+    public function __construct()
+    {
+        $this->format = ['csv','json'];
+    }
+
+    /**
+     * check $format and $method and lead to the function choseFormat
+     * @param $format
+     * @param null $id
+     * @return array
+     * @throws \Exception
+     */
+    public function apiService($format, $id)
+    {
+        $this->checkFormat($format);
+        return $this->choseFormat($format, $id);
+    }
+
+    /**
+     * check $format
+     * @param $format
+     * @throws \Exception
+     */
+    private function checkFormat($format)
+    {
+        if (!in_array($format, $this->format))
+        {
+            throw new \Exception("format invalid");
+        }
+    }
+
+    /**
+     * @param $format
+     * @param $id
+     * @return array|mixed
+     * @throws \Exception
+     */
+    private function choseFormat($format, $id)
+    {
+        if ($format == 'json') {
+            $formatInstance = new Json();;
+        } elseif ($format == 'csv') {
+            $formatInstance = new Csv();
+        }
+        $method = $format . 'Method';
```

Comment:
This approach can be bit ugly but as long you will learn more about OOP Interfaces and Classes inheritance, this can be solve very easy and have something very intuitive. I will describe what you can do, it's not necessary to change but just to have an idea.
```
interface Extractor {
    public function extract($id);
}
class Json implements Extractor {
    public function extract($id) {
     //logic to extract json...
    }
}
class Csv implements Extractor {
    public function extract($id) {
     //logic to extract csv...
    }
}
```
with this approach our code will look like this:
```
       if ($format == 'json') {
            $formatInstance = new Json();
        } elseif ($format == 'csv') {
            $formatInstance = new Csv();
        }
       return $formatInstance->extract($id);
```

Reply:
I propose to scope this refactor to in a separated PR because it's a 2.0.0 change